### PR TITLE
Task/bws 185 pass arguments to the function custom popup

### PR DESCRIPTION
**Pull request by junojei**
1280x720

https://user-images.githubusercontent.com/96128765/160245601-17c1efb4-ef13-4395-ad50-a569fda54b80.mp4

### CODE REVIEW

File: frontend/src/components/slc-configurate-function/components/edit-form/components/slc-popup/slc-popup.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        <h4 className={styles.title}>{'Pass function arguments'}</h4>
+        <Editor
+          value={value}
+          onChangeValue={handleOnChangeValue}
+          lang={EditorLang.JSON}
+          placeholder={
+            'Enter your arguments in JSON format.(optional)\nExample: { "key": "value" }'
+          }
+          height={300}
+        ></Editor>
```
Comment (what1s1ove) [inline frontend/src/components/slc-configurate-function/components/edit-form/components/slc-popup/slc-popup.tsx]:
```suggestion
        />
```

File: frontend/src/components/slc-configurate-function/components/edit-form/components/slc-popup/slc-popup.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            className={styles.button}
+            label="Cancel"
+            onClick={onClose}
+          ></Button>
+          <Button
+            className={styles.button}
+            btnStyle={ButtonStyle.FILLED}
+            label="Run"
+            onClick={handleOnRun}
+          ></Button>
```
Comment (what1s1ove) [inline frontend/src/components/slc-configurate-function/components/edit-form/components/slc-popup/slc-popup.tsx]:
```suggestion
          />
          <Button
            className={styles.button}
            btnStyle={ButtonStyle.FILLED}
            label="Run"
            onClick={handleOnRun}
          />
```

File: frontend/src/components/slc-configurate-function/components/edit-form/styles.module.scss [hunk 3]
```diff
@@ -18,7 +18,7 @@
 }
 
 .button {
-  min-width: 100px;
+  width: 100px;
```
Comment (what1s1ove) [inline frontend/src/components/slc-configurate-function/components/edit-form/styles.module.scss]: why?
left `min-width`

File: frontend/src/components/slc-configurate-function/components/edit-form/edit-form.tsx [hunk 4]
```diff
@@ -102,17 +113,28 @@ const EditForm: FC<Props> = ({ id }) => {
...
+          onClick={handleCancel}
+        />
       </div>
+      {isVisibleSLCPopup && (
+        <SLCPopup
+          isOpen={isVisibleSLCPopup}
+          onClose={handleCloseSLCPopup}
+          onRun={handleRun}
+        />
+      )}
```
Comment (what1s1ove) [inline frontend/src/components/slc-configurate-function/components/edit-form/edit-form.tsx]:
```suggestion
        <SLCPopup
          isOpen={isVisibleSLCPopup}
          onClose={handleCloseSLCPopup}
          onRun={handleRun}
        />
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/components/slc-configurate-function/components/edit-form/edit-form.tsx (hunk 4 above)
Comment (junojei) [reply frontend/src/components/slc-configurate-function/components/edit-form/edit-form.tsx]: I do it for reset the component

File: frontend/src/components/slc-configurate-function/components/edit-form/components/slc-popup/slc-popup.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  const [value, setValue] = useState<string>('');
+
+  useEffect(() => {
+    setValue('');
+  }, [isOpen]);
+
+  const handleOnChangeValue = debounce(setValue, TIMEOUT);
+
+  const handleOnRun = (): void => {
+    onRun(value);
```
Comment (what1s1ove) [inline frontend/src/components/slc-configurate-function/components/edit-form/components/slc-popup/slc-popup.tsx:26]: what if an error occurs? you need to handle it. check redux toolkit documentation

**Review by ania-leichenko — DISMISSED** [review]

**Review by Yaroslaww-1 — DISMISSED** [review]

File: frontend/src/components/slc-configurate-function/components/edit-form/components/slc-popup/slc-popup.tsx (hunk 5 above)
Comment (what1s1ove) [reply frontend/src/components/slc-configurate-function/components/edit-form/components/slc-popup/slc-popup.tsx:26]: request after fix

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**